packages/mdx/lib/plugin/recma-document.js

```diff
@@ -518,7 +524,21 @@ export function recmaDocument(options = {}) {
     }
 
     // @ts-expect-error: JSXElements are expressions.
-    const consequent = /** @type {Expression} */ (element)
+    let consequent = /** @type {Expression} */ (element)
```

Comment:
Maybe a different name, now that this no longer has to be a consequent. `result`?